hls-plugin-api/src/Ide/Types.hs

```diff
@@ -1051,30 +1054,48 @@ mkCodeActionHandlerWithResolve codeActionMethod codeResolveMethod =
 -- support. This means you don't have to check whether the client supports resolve
 -- and act accordingly in your own providers.
 mkCodeActionWithResolveAndCommand
-  :: forall ideState. (ideState -> PluginId -> CodeActionParams -> LspM Config (Either ResponseError ([Command |? CodeAction] |? Null)))
+  :: forall ideState.
+  PluginId
+  -> (ideState -> PluginId -> CodeActionParams -> LspM Config (Either ResponseError ([Command |? CodeAction] |? Null)))
   -> (ideState -> PluginId -> CodeAction -> LspM Config (Either ResponseError CodeAction))
-  -> PluginHandlers ideState
-mkCodeActionWithResolveAndCommand codeActionMethod codeResolveMethod =
+  -> ([PluginCommand ideState], PluginHandlers ideState)
+mkCodeActionWithResolveAndCommand plId codeActionMethod codeResolveMethod =
     let newCodeActionMethod ideState pid params = runExceptT $
             do codeActionReturn <- ExceptT $ codeActionMethod ideState pid params
                caps <- lift getClientCapabilities
                case codeActionReturn of
                 r@(InR Null) -> pure r
                 (InL ls) | -- If the client supports resolve, we will wrap the resolve data in a owned
                            -- resolve data type to allow the server to know who to send the resolve request to
-                           -- and dump the command fields.
                            supportsCodeActionResolve caps ->
-                            pure $ InL (dropCommands . wrapCodeActionResolveData pid <$> ls)
-                           -- If they do not we will drop the data field.
-                         | otherwise -> pure $ InL $ dropData <$> ls
+                            pure $ InL (wrapCodeActionResolveData pid <$> ls)
+                           -- If they do not we will drop the data field, in addition we will populate the command
+                           -- field with our command to execute the resolve, with the whole code action as it's argument.
+                         | otherwise -> pure $ InL $ moveDataToCommand <$> ls
         newCodeResolveMethod ideState pid params =
             codeResolveMethod ideState pid (unwrapCodeActionResolveData params)
-    in mkPluginHandler SMethod_TextDocumentCodeAction newCodeActionMethod
-    <> mkPluginHandler SMethod_CodeActionResolve newCodeResolveMethod
-  where dropData :: Command |? CodeAction -> Command |? CodeAction
-        dropData ca = ca & _R . L.data_ .~ Nothing
-        dropCommands :: Command |? CodeAction -> Command |? CodeAction
-        dropCommands ca = ca & _R . L.command .~ Nothing
+    in ([PluginCommand "codeActionResolve" "Executes resolve for code action" (executeResolveCmd plId codeResolveMethod)],
```

Comment:
Hmm, I'm not sure if this will work? This seems to use the same command ID for all the resolve commands. I'm not sure what will happen then! Could we disambiguate the command ID using the plugin ID?

Reply:
I believe it already "magically" does this for us

Reply:
Ah yes, you're right. Could you add some haddock maybe?